ragna/core/_rag.py

```diff
@@ -218,7 +239,10 @@ async def answer(self, prompt: str, *, stream: bool = False) -> Message:
 
         self._messages.append(Message(content=prompt, role=MessageRole.USER))
 
-        sources = await self._run(self.source_storage.retrieve, self.documents, prompt)
+        if list[Document] in inspect.signature(self.source_storage.store).parameters.values():
```

Comment:
This hits a point that I didn't consider before: we are currently passing the documents again to the retrieve function. See the part about BC in https://github.com/Quansight/ragna/issues/256#issuecomment-1954057313 for a reason why. This will likely change when we implement #256. However, in the mean time we need to decide if we want the same "input switching" here as for store. I think this is ok, but want to hear your thoughts.

Reply:
Ooh, I only in hindsight understand the logic here. Of course we need to be able to embed the prompt. So this correct.